### ISSUE TITLE
Unable to install library edamov/pushok

### ISSUE THREAD

**Issue by fbartolom** [edamov/pushok#80]
I keep receiving error 

> - edamov/pushok 0.6.4 requires lib-curl >=7.46.0 -> the requested linked library curl has the wrong version installed or is missing from your system, make sure to have the extension providing it.

when I run `composer require edamov/pushok` notwithstanding I have lib-curl at version 7.66 as seen by:
```
curl-config --version
libcurl 7.66.0
```
May you check?
Thanks.

**Comment by edamov** [comment]
@fbartolom Is php compiled with that version of cURL?

**Comment by fbartolom** [comment]
I was not aware that was needed: how do I check and possibly fix it?

Thanks

Dr. Fabrizio Bartolomucci
eMail: fbartolom@gmail.com
iPhone: +39 3474013121

**Comment by fbartolom** [comment]
At any rate my php version is quite new:
PHP 7.3.10 (cli) (built: Sep 24 2019 09:20:18)

**Comment by edamov** [comment]
You can check it in phpinfo().

**Comment by fbartolom** [comment]
I would not know what to search, at any rate I loaded it on the web space of the server and you may check it yourself at https://taxiprofessional.net/phpinfo.php <https://taxiprofessional.net/phpinfo.php>

**Comment by edamov** [comment]
You have curl 7.29.0 - http://i.imgur.com/Y2gCeN6.png
You need to recompile php with new version of curl

**Comment by fbartolom** [comment]
Yes, but I thought from your error message that it was the libcurl to be at the version you indicated:
edamov/pushok 0.6.4 requires lib-curl >=7.46.0
not, curl, also given I have no cue about how to update it, I tried everything and only managed to update lib-curl to 7.66.
If it is also curl to be upgraded, please fix your error message and tell me a reliable way for updating it: yum update php does nothing as well all other the attempts I tried.

**Comment by edamov** [comment]
This is not error message from pushok library. This error message is generated by composer.

I don't know how to compile php using yum.
Maybe this will help you - https://stackoverflow.com/questions/4976971/compiling-php-with-curl-where-is-curl-installed

**Comment by fbartolom** [comment]
That post also specifies how to update libcurl, what I managed. I found post (https://tecadmin.net/install-php7-on-centos7/) by which I was able to get to version 7.4.0: yet trying to execute: `yum --enablerepo=remi-php746 install php` or `yum --enablerepo=remi-php75` finds nothing like whether 7.4.0 were the last available version for Centos 7, what seems still not enough for your very picky library.

**Comment by fbartolom** [comment]
It also seems the latest version of php for all platforms be 7.4.0, as you may check yourself on the web; I do not think it is an error the one of the composer command to request version 7.4.6 for libcurl, perhaps it is an error the fact that it does not find it, notwithstanding it is available. Please check if you may investigate the issue.

**Comment by fbartolom** [comment]
Now, after restarting Apache, php got to version 7.6.6 and still composer does not work:
curl -V
curl 7.66.0 (x86_64-pc-linux-gnu) libcurl/7.66.0 OpenSSL/1.0.2k-fips zlib/1.2.7
Release-Date: 2019-09-11
Protocols: dict file ftp ftps gopher http https imap imaps pop3 pop3s rtsp smb smbs smtp smtps telnet tftp 
Features: AsynchDNS HTTPS-proxy IPv6 Largefile libz NTLM NTLM_WB SSL UnixSockets

**Comment by fbartolom** [comment]
And for good measure also libcurl has been updated:
```
curl-config --version
libcurl 7.66.0
```
and still composer return an error.

**Comment by edamov** [comment]
You are confusing php v. 7.4.0 and lib-curl v. 7.4.6. It is not the same.
And also php can have another version of lib-curl (unlike the version installed on your system).
When you run `curl -V` you see the version of curl library in your system. But php can be compiled with another version of lib-curl (you can check in phpinfo()).
You can use [docker image](https://hub.docker.com/r/edamov/pushok-docker/dockerfile). It has required versions of all dependencies.
Also please read this [answer](https://stackoverflow.com/questions/41370391/how-to-compile-php-with-specific-versions-of-curl-and-openssl). Maybe it will help.

**Comment by fbartolom** [comment]
The point is the the highest version available is the one I have installed. There doesn’t exist a higher version than 7.4.0 altogether; it just is the curllib to have reached version 7.4.6. If you know better, please let me know where I may find version 7.4.6, otherwise it is like talking of elves and goblins... 

Dr. Fabrizio Bartolomucci
eMail: fbartolom@gmail.com
iPhone: +39 3474013121

**Comment by edamov** [comment]
You installed 7.4.0 - this is php version.
7.4.6 - is the lib-curl version. This is not php version.
For example, you can use `php 7.1.0` and `lib-curl 7.46`.
Or you can use `php 7.2.0` and `lib-curl 7.46`.
There are a lot of lib-curl [versions](https://curl.haxx.se/docs/releases.html) and the minimum required is `7.46`.
So you can use your php 7.4.0 and lib-curl 7.66 (or any other but greater than 7.46) 
[Here](https://taxiprofessional.net/phpinfo.php)  you have lib-curl 7.29.0. It will not work for this library.

**Comment by fbartolom** [comment]
Actually if I execute:
curl-config —version
I get:
libcurl 7.66.0
Not 7.29.0

Not sure where you did find I have that lesser version.

Dr. Fabrizio Bartolomucci
eMail: fbartolom@gmail.com
iPhone: +39 3474013121

**Comment by ozgurhangisi** [comment]
@fbartolom there are 2 points. You have to have the latest curl version on your machine and you have to compile your php with the latest curl version. You can have curl 7.66.0 but it's look like you compiled your php with curl 7.29.0 Please check your phpinfo ( https://taxiprofessional.net/phpinfo.php )  and search for cURL Information. You are going to see curl 7.29.0.

So the problem you have is not about this library. It is about your php installation.

**Comment by fbartolom** [comment]
The reason why curl -V gives a result and phpinfo another is unknown to me; yet I cannot update a version already updated, given from the OS point of view I already have 7.4.0. What, to my knowledge, is the last delivered version.

**Comment by fbartolom** [comment]
At any rate now php reports 7.4 even on php-info, how you may check yourself. And still composer does not start.

**Comment by edamov** [comment]
curl -V and phpinfo give you different versions because php is compiled with curl 7.29
You just need to recompile php with a newer version of curl

**Comment by fbartolom** [comment]
The question is how: I already disinstalled and reinstalled php, but that changed nothing.

**Comment by fbartolom** [comment]
I apparently finally manages to install curl 7.66, as you may check on php-info, and in fact, after installing the required php-gmp library, the composer error changed to:

Your requirements could not be resolved to an installable set of packages.
Problem 1
    - Installation request for edamov/pushok ^0.6.4 -> satisfiable by edamov/pushok[0.6.4].
    - edamov/pushok 0.6.4 requires ext-intl * -> the requested PHP extension intl is missing from your system.

  To enable extensions, verify that they are enabled in your .ini files:
    - /etc/php.ini
    - /etc/php.d/20-bcmath.ini
    - /etc/php.d/20-bz2.ini
    - /etc/php.d/20-calendar.ini
    - /etc/php.d/20-ctype.ini
    - /etc/php.d/20-curl.ini
    - /etc/php.d/20-dom.ini
    - /etc/php.d/20-exif.ini
    - /etc/php.d/20-fileinfo.ini
    - /etc/php.d/20-ftp.ini
    - /etc/php.d/20-gd.ini
    - /etc/php.d/20-gettext.ini
    - /etc/php.d/20-gmp.ini
    - /etc/php.d/20-iconv.ini
    - /etc/php.d/20-json.ini
    - /etc/php.d/20-mbstring.ini
    - /etc/php.d/20-mysqlnd.ini
    - /etc/php.d/20-pdo.ini
    - /etc/php.d/20-phar.ini
    - /etc/php.d/20-posix.ini
    - /etc/php.d/20-shmop.ini
    - /etc/php.d/20-simplexml.ini
    - /etc/php.d/20-sockets.ini
    - /etc/php.d/20-sqlite3.ini
    - /etc/php.d/20-sysvmsg.ini
    - /etc/php.d/20-sysvsem.ini
    - /etc/php.d/20-sysvshm.ini
    - /etc/php.d/20-tokenizer.ini
    - /etc/php.d/20-xml.ini
    - /etc/php.d/20-xmlwriter.ini
    - /etc/php.d/20-xsl.ini
    - /etc/php.d/30-mcrypt.ini
    - /etc/php.d/30-mysqli.ini
    - /etc/php.d/30-pdo_mysql.ini
    - /etc/php.d/30-pdo_sqlite.ini
    - /etc/php.d/30-xmlreader.ini
  You can also run `php --ini` inside terminal to see which files are used by PHP in CLI mode.
I installed php74-php-intl.x86_64, but that changed nothing.

What should I do now?

Dr. Fabrizio Bartolomucci
eMail: fbartolom@gmail.com
iPhone: +39 3474013121

**Comment by edamov** [comment]
Please check phpinfo() again. Do you have intl [section](http://i.imgur.com/ZmaQSgv.png)?

**Comment by fbartolom** [comment]
I did and it is not there in fact: notwithstanding I installed php74-php-intl.x86_64:

Il pacchetto php74-php-intl-7.4.0~rc4-29.el7.remi.x86_64 è già installato e aggiornato all'ultima versione
Niente da fare

What should I install instead? The server is on centos 7.

**Comment by edamov** [comment]
Maybe [this](https://unix.stackexchange.com/a/345252) answer will help you

**Comment by fbartolom** [comment]
That was in fact the post I perused to install the Intl library I have: in fact if I execute those commands again I get:
Il pacchetto php-intl-7.4.0~RC4-10.el7.remi.x86_64 è già installato e aggiornato all'ultima versione
Meaning there is nothing to do given the SW is already installed.
Of course I tried many things before contacting you again.

Dr. Fabrizio Bartolomucci
eMail: fbartolom@gmail.com
iPhone: +39 3474013121
